### PR TITLE
Refactor/2318/migrate area settings panel

### PR DESCRIPTION
**Pull request by shaman-apprentice**
migrate area settings panel

### CODE REVIEW

File: visualization/app/codeCharta/ui/ribbonBar/areaSettingsPanel/areaSettingsPanel.component.html [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+></cc-slider>
+
+<div class="bottom-row">
+	<mat-checkbox
+		class="default-margin"
+		[checked]="dynamicMargin$ | async"
+		(change)="setDynamicMargin($event)"
+		title="Set margin to default (50px)."
+		>Default margin (50px)</mat-checkbox
+	>
```
Comment (shaman-apprentice) [inline visualization/app/codeCharta/ui/ribbonBar/areaSettingsPanel/areaSettingsPanel.component.html:17]:
```suggestion
	<mat-checkbox
		class="default-margin"
		[checked]="dynamicMargin$ | async"
		(change)="setDynamicMargin($event)"
		title="Set margin to default (50px)."
	>
		Default margin (50px)
	</mat-checkbox>
```

File: visualization/app/codeCharta/ui/ribbonBar/areaSettingsPanel/invertAreaOption/invertAreaOption.component.ts [hunk 2]
```diff
@@ -1,15 +1,15 @@
...
+import { toggleInvertingArea } from "../../../../state/store/appSettings/invertArea/invertArea.actions"
 import { Observable } from "rxjs"
-import { toggleInvertingArea } from "../../../state/store/appSettings/invertArea/invertArea.actions"
 
 @Component({
 	selector: "cc-invert-area-option",
 	template: require("./invertAreaOption.component.html")
 })
-export class InvertAreaOptionComponent implements OnInit {
+export class InvertAreaOptionComponent {
```
Comment (shaman-apprentice) [inline visualization/app/codeCharta/ui/ribbonBar/areaSettingsPanel/invertAreaOption/invertAreaOption.component.ts:12]: this still does implement `OnInit` - depending on compile process this actually may lead to ngOnInit not being executed.

File: visualization/app/codeCharta/ui/ribbonBar/areaSettingsPanel/areaSettingsPanel.component.html (hunk 1 above)
Comment (IhsenBouallegue) [reply visualization/app/codeCharta/ui/ribbonBar/areaSettingsPanel/areaSettingsPanel.component.html]: resolved in the next commit

File: visualization/app/codeCharta/ui/ribbonBar/areaSettingsPanel/invertAreaOption/invertAreaOption.component.ts (hunk 2 above)
Comment (IhsenBouallegue) [reply visualization/app/codeCharta/ui/ribbonBar/areaSettingsPanel/invertAreaOption/invertAreaOption.component.ts]: Resolved in the next commits

File: visualization/app/codeCharta/ui/ribbonBar/areaSettingsPanel/areaSettingsPanel.component.spec.ts [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+describe("AreaSettingsPanelComponent", () => {
+	beforeEach(() => {
+		Store["initialize"]()
+		TestBed.configureTestingModule({
+			imports: [AreaSettingsPanelModule]
+		})
+	})
+
+	it("should handle default margin", async function () {
```
[the remark this answers is not in the document]
Comment (IhsenBouallegue) [reply visualization/app/codeCharta/ui/ribbonBar/areaSettingsPanel/areaSettingsPanel.component.spec.ts]: Resolved in the next commits

**Review by IhsenBouallegue — APPROVED** [review]
LGTM 🥳

**Pull request merged**